services/meta/client.go

```diff
@@ -694,11 +694,19 @@ func (c *Client) DropShard(id uint64) error {
 
 // CreateShardGroup creates a shard group on a database and policy for a given timestamp.
 func (c *Client) CreateShardGroup(database, policy string, timestamp time.Time) (*ShardGroupInfo, error) {
+	// Check under a read-lock
```

Comment:
I think a little more detail explaining the reason for the optimistic check under read lock would be worthwhile.